src/distrx/transforms.py

```diff
@@ -9,33 +9,169 @@
 * Add decorators for accepting floats or vectors
 
 """
-from typing import Tuple
+
 import warnings
+from typing import Tuple
 
 import numpy as np
 import numpy.typing as npt
 
 
-TRANSFORM_DICT = {
-    'log': [
-        np.log,
-        lambda x: 1.0/x
-    ], 'logit': [
-        lambda x: np.log(x/(1.0 - x)),
-        lambda x: 1.0/(x*(1.0 - x))
-    ], 'exp': [
-        np.exp,
-        np.exp
-    ], 'expit': [
-        lambda x: 1.0/(1.0 + np.exp(-x)),
-        lambda x: np.exp(-x)/(1.0 + np.exp(-x))**2
-    ]
-}
-METHOD_LIST = ['delta']
-
-
-def transform_data(mu: npt.ArrayLike, sigma: npt.ArrayLike, transform: str,
-                   method: str = 'delta') -> Tuple[np.ndarray, np.ndarray]:
+class FirstOrder:
```

Comment:
Hi @mbi6245!
Usually we don't use the initialization function of the class as the main call. It will make the instance not reusable. I think a better way to do this is to only pass in the `transform` in the init. And the instance will determine which branch of the function it will use.

For example
```python
class FirstOrder:
    def __init__(self, transform: str) -> None:
        # you can raise error if `transform` is not one of the options
        self.transform = transform

    def __call__(self, mu: npt.ArrayLike, sigma: npt.ArrayLike) -> tuple[npt.NDArray, npt.NDArray]:
        return getattr(self, f"{self.transform}_trans")(mu, sigma)
```